dace/codegen/targets/rtl.py

```diff
@@ -96,8 +97,17 @@ def copy_memory(self, sdfg: sdfg.SDFG, dfg: state.StateSubgraphView, state_id: i
                                                               dst_node.in_connectors[edge.dst_conn].ctype,
                                                               edge.src.data)
             else:  # scalar accessor
-                line: str = "{}* {} = &{}[0];".format(dst_node.in_connectors[edge.dst_conn].ctype, edge.dst_conn,
-                                                      edge.src.data)
+                arr = sdfg.arrays[edge.data.data]
+                if isinstance(arr, data.Array):
+                    line: str = "{}* {} = &{}[0];".format(dst_node.in_connectors[edge.dst_conn].ctype, edge.dst_conn,
+                                                          edge.src.data)
+                elif isinstance(arr, data.Scalar):
+                    line: str = "{} {} = {};".format(dst_node.in_connectors[edge.dst_conn].ctype, edge.dst_conn,
+                                                     edge.src.data)
+        elif isinstance(edge.src, nodes.MapEntry) and isinstance(edge.dst, nodes.Tasklet):
+            # TODO works for one kernel, but might break with several. Also only handles if the scope and tasklet are directly connected.
+            self.n_unrolled = symbolic.evaluate(edge.src.map.range[0][1] + 1, sdfg.constants)
```

Comment:
Why does it break with multiple kernels? If it's because of the `self.n_unrolled`, you can make this a dict instead or something, so you can assign it differently for different kernels

Reply:
The thought arose when implementing: this could fail if there was multiple kernels, since `n_unrolled` is bound to the codegen, although everything inside might not be unrolled. I can't think of an example, but converting to a dictionary sounds like a safe move.

Reply:
This has been implemented.